src/native-stack/utils/usePreventDismiss.tsx

```diff
@@ -0,0 +1,15 @@
+import * as React from 'react';
+
+import PreventDismissContext from './PreventDismissContext';
+
+export default function usePreventDismiss() {
+  const preventDismiss = React.useContext(PreventDismissContext);
+
+  if (preventDismiss === undefined) {
+    throw new Error(
+      "Couldn't find value for preventing dismiss. Are you inside a screen in Native Stack?"
+    );
+  }
+
+  return preventDismiss;
+}
```

Comment:
The idea is that user would use this hook to declaratively enable/disable preventing dismiss. So they would pass the option to the hook, and not call `preventDismiss` manually. We should call it automatically in an effect.

Next thing is that this hook could be used in multiple components. Having a boolean for this state won't work since multiple of them would conflict. We'd need a symbol or ID to identify it.

```suggestion
type Options = {
  enabled: boolean
}

export default function usePreventDismiss({ enabled }: Options) {
  const [symbol] = React.useState(() => Symbol());
  const [setPreventDismiss, removePreventDismiss] = React.useContext(PreventDismissContext);

  if (preventDismiss === undefined) {
    throw new Error(
      "Couldn't find value for preventing dismiss. Are you inside a screen in Native Stack?"
    );
  }
  
  React.useEffect(() => {
     setPreventDismiss(symbol, enabled);
    
    return () => removePreventDismiss(symbol);
  }, [symbol, enabled]);
}
```